stacks/api/main.ts

```diff
@@ -1,23 +1,184 @@
+import * as yargs from "yargs";
 
-async function bootstrap() {
-  let httpsOptions: https.ServerOptions;
+const args = yargs
+  /* TLS Options */
+  .options({
+    "cert-file": {
+      string: true,
+      description: "Path to the client server TLS cert file."
+    },
+    "key-file": {
+      string: true,
+      description: "Path to the client server TLS key file."
+    }
+  })
+  /* Database Options */
+  .options({
+    "database-uri": {
+      string: true,
+      description: "MongoDB connection url."
+    },
+    "database-name": {
+      string: true,
+      description: "Name of the database."
+    },
+    "database-replica-set": {
+      string: true,
+      alias: ["replica-set"],
+      description: "Name of the replica set."
+    },
+    "database-pool-size": {
+      number: true,
+      description: "Amount of connection that will be opened against database.",
+      default: 100
+    }
+  })
+  .demandOption("database-name")
+  .demandOption("database-uri")
+  /* Feature Toggling: Bucket and Activity Stream */
+  .options({
+    "activity-stream": {
+      boolean: true,
+      description: "Whether Activity Stream feature is enabled.",
+      default: true
+    },
+    "bucket-hooks": {
+      boolean: true,
+      description: "Whether Bucket Hooks feature is enabled.",
+      default: true
+    },
+    "bucket-history": {
+      boolean: true,
+      description: "Whether Bucket History feature is enabled.",
+      default: true
+    },
+    "experimental-bucket-realtime": {
+      boolean: true,
+      description: "Whether Experimental Bucket Realtime feature is enabled.",
+      default: true
+    }
+  })
+  /* Passport Options  */
+  .options({
+    "passport-secret": {
+      string: true,
+      alias: ["secret"],
+      description: "PEM-encoded public key (asymmetric) for verifying the JWT token's signature."
+    },
+    "passport-password": {
+      string: true,
+      description: "Password of the default 'spica' account.",
+      default: "spica"
+    }
+  })
+  .demandOption("passport-secret")
+  /* Function Options */
+  .options({
+    "function-pool-size": {
+      number: true,
+      description: "Number of worker processes to fork at start up.",
+      default: 10
+    }
+  })
+  /* Common Options */
+  .option("port", {
+    number: true,
+    alias: ["p"],
+    default: 80,
+    description: "A port to listen on."
+  })
+  .option("persistent-path", {
+    string: true,
+    default: "/var/data",
+    normalize: true,
+    description: "A path that will be used by API to store persistent data."
+  })
+  .option("public-url", {
+    string: true,
+    description: "Publicly accessible url of the APIs.",
+    alias: ["public-host"],
+    describe: `This option will be used by various modules. 
+For instance the 'storage' module will use this url to generate object links that are accessible from egress networks
+While the 'passport' module will use this url to re-route the user to the Passport APIs.

```

Comment:
Don't we need to put a default value for this public-url like 'http://localhost:4300'?

Reply:
Our development defaults come from .bazelrc through the environment variables. So we don't need to put a default. instead, we demand this argument to be filled by the invoker.